### DISCUSSION TITLE
Material.Avalonia not working

### DISCUSSION THREAD

**Discussion by AvihayBarn** [AvaloniaCommunity/Material.Avalonia#404] [Q&A] [answered]
Hello , 
I would love to get help. 
I develop an application with Avalonia version 11.1.0 and when i try to "import" styles from it and use the 
xmlns:wpf="clr-namespace:Material.Styles.Assists;assembly=Material.Styles" when i write wpf: i cant find all the assists and also 
xmlns:styles="clr-namespace:Material.Styles;assembly=Material.Styles" when i adding class for exmaple to TextbOX IN Classes="Outline" nothing happen. I searched for similar issue but i could not find. 
thank you very much for any help in advance.

**Comment by SKProCH** [comment, marked as answer]
Hello. 
Assists located in the `Material.Styles.Assists` namespace, so its should be correct.  
Outline class should be lowercase like `Classes="outline"`, but its deprecated, so prefer using `Theme="{StaticResource OutlineTextBox}"`.
Also, this imports required only for compiler to understand correct location of classes in code. Themes must be imported separately in App.axaml, make sure what add `MaterialTheme` into App.axaml.

Also, make sure what you are using latest version of this library.

You can run demo app (locally or using wasm version in browser) and see all code required for any control in demo by clicking code button in left bottom corner of every control.

**Reply by AvihayBarn** [reply]
Thank you very much! The problem was that i didnt import the Theme in the App.xaml file. I think that I didnt understand the mechanism of it. There is a way to stay with the default avalonia theme and still import just one specific style? By the way, in the demo project i saw that you use classes="Outline" with capital O.

**Reply by SKProCH** [reply]
Yes, you can import single file with `<ResourceInclude Source="avares://Material.Styles/Controls/Card.axaml" />`, but most of controls still relies on theme colors, so probably you still need an `MaterialTheme`.

This is probably XY problem, so you should describe what you want to achieve. If you want to use mostly Fluent theme, but some controls with material theme, you can add `MaterialTheme` BEFORE the `FluentTheme` in App.axaml and explicitly set theme for certain controls using `Theme="{StaticResource OutlineTextBox}"`.

> classes="Outline"

In older version - probably, but in current repo I can't find such examples. If you find something like that - please, let me know. 
P.S. Anyway, the classes approach is a legacy from the time when Themes isn't added to Avalonia, so currently Theme is preferred way to select control *theme*. Classes still supported as a backward compatibility feature.

**Reply by SKProCH** [reply]
> I think that I didnt understand the mechanism of it

Avalonia will start finding required resources (unset Theme for control == type(Control)) up to visual tree in you file (e.g. in MainWindow). If nothing found it will continue to lookup the App.Resources (which are defined in App.axaml) from bottom to top. FluentTheme and MaterialTheme is basically an ResourceDictionary with extra layer of configuration.

So, adding Theme's to App.axaml is important for Avalonia to find the actual control themes.
